core/server/api/authentication.js

```diff
@@ -321,7 +321,13 @@ authentication = {
                     updatedUser.set('status', 'active');
                     return updatedUser.save(options);
                 })
+                .catch(errors.ValidationError, function (err) {
+                    return Promise.reject(err);
+                })
                 .catch(function (err) {
+                    if (errors.utils.isIgnitionError(err)) {
+                        return Promise.reject(err);
+                    }
                     throw new errors.UnauthorizedError({err: err});
```

Comment:
Nitpick / note: `return Promise.reject(err)` and `throw new err` are 2 ways to do the same thing. Most of the time we use the first one, maybe swap out `throw new` here?